core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -1970,6 +1980,8 @@ public static void registerDefaultRules(RelOptPlanner planner,
     // Change the below to enable constant-reduction.
     if (false) {
       registerReductionRules(planner);
+    } else {
+      planner.addRule(ReduceExpressionsRule.FILTER_INSTANCE);
```

Comment:
I suggested enabling this rule but I didn't remember that all the other constant reduction rules were disabled. Does somebody know why we disabled explicitly the reduction rules?

Reply:
My bet would be that we prefer to do constant reduction in other places and not during planning. However as this case shows it might make sense to have also certain reduction rules enabled during planning. Any opinions?

Reply:
Indeed the cases that need constant reduction in rule planning is rare, except for the cases we bring in new constant in the planning rules.

In Flink, we have a little such rules to explode some predicates for Join condition(in order to generate HashJoin), but indeed, the benefits of constant reduction are small.

Reply:
The logic looks weird, isn't the `ReduceExpressionsRule.FILTER_INSTANCE` also belongs to reduction rules ?

Reply:
yes, it belongs. At least, here it will be visible that after enabling each reduction rule one by one in future the ```if(false)``` could be safely replaced by ```registerReductionRules(planner);```.

Reply:
If we decide to enable `ReduceExpressionsRule.FILTER_INSTANCE` then we should definitely refactor this piece of code possibly keeping only `planner.addRule(ReduceExpressionsRule.FILTER_INSTANCE);`

Reply:
done